src/backend/aspen/api/auth.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+import sqlalchemy as sa
+from auth0.v3.exceptions import TokenValidationError
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import joinedload
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy.orm.query import Query
+from starlette.requests import Request
+
+from aspen.api.config.config import settings
+from aspen.api.deps import get_db
+from aspen.auth.device_auth import validate_auth_header
+from aspen.database.models import Group, User
+
+
+def get_usergroup_query(session: AsyncSession, user_id: str) -> Query:
+    return (
+        sa.select(User)  # type: ignore
+        .options(joinedload(User.group).joinedload(Group.can_see))  # type: ignore
+        .filter(User.auth0_user_id == user_id)  # type: ignore
+    )
```

Comment:
This is SQLAlchemy's 2.0 preferred query syntax. I haven't figured out how to make it play nice with mypy yet, but that will need to happen soon since fastapi will need to start executing more queries!